### ISSUE TITLE
Xcode compilation error with 1.7.2

### ISSUE THREAD

**Issue by 3DExtended** [flame-engine/flame#2499]
# Current bug behavior
I have upgraded my flutter project and receiving an error while compiling in Xcode (versions see below).
The error states: 

```
Failed to build iOS app
Error (Xcode): ../../../../../.pub-cache/hosted/pub.dev/flame-1.7.2/lib/src/components/core/recycled_queue.dart:23:7: Error: 'Iterable' can't be used in both 'extends' and
'implements' clauses.
```

# Expected behavior
I would expect, that this error does not get thrown.

# Steps to reproduce
Sadly, I am not sure how to reproduce. Maybe the versions are a better clue.

# Flutter doctor output
```
[✓] Flutter (Channel master, 3.10.0-5.0.pre.33, on macOS 13.3.1 22E261 darwin-x64, locale de-DE)
    • Flutter version 3.10.0-5.0.pre.33 on channel master at /usr/local/Caskroom/flutter/3.0.5/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b95cd931ae (38 minutes ago), 2023-04-14 13:20:49 -0400
    • Engine revision ea0e1b1700
    • Dart version 3.1.0 (build 3.1.0-4.0.dev)
    • DevTools version 2.23.1

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.76.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] VS Code (version 1.72.0-insider)
    • VS Code at /Applications/Visual Studio Code - Insiders.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.3.1 22E261 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 112.0.5615.49

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.

```

# More environment information

* Flame version: 1.7.2
* Platform affected: ios
* XCode: 14.3
* Flutter 3.10.0-5.0.pre.33 • channel master • https://github.com/flutter/flutter.git
* Framework • revision b95cd931ae (41 minutes ago) • 2023-04-14 13:20:49 -0400
* Engine • revision ea0e1b1700
* Tools • Dart 3.1.0 (build 3.1.0-4.0.dev) • DevTools 2.23.1

# More information
I recently upgrade from version 1.6.0 to 1.7.2 (along with other dependency updates). 
Maybe this issue is related to the upgrade from dart: ">=2.19.0 <4.0.0" to dart: ">=3.0.0-134.0.dev <4.0.0".

I hope you can help me!

**Comment by spydon** [comment]
It's because you're on Flutter beta or master, switch to Flutter stable and it will work. :)

**Comment by 3DExtended** [comment]
Sadly, this error remains with version 1.7.0:

```Error (Xcode): ../../../../../.pub-cache/hosted/pub.dev/flame-1.7.0/lib/src/components/core/recycled_queue.dart:23:7: Error: 'Iterable' can't be used in both 'extends' and 'implements' clauses.```

**Comment by 3DExtended** [comment]
Downgrading to 1.6.0 works for now... not sure if you want to close this bug or need further details for fixing it. please let me know

**Comment by spydon** [comment]
It's not Flame that you have to downgrade, it's Flutter.

**Comment by spydon** [comment]
This is solved to work with Flutter Beta in Flame v1.7.3
